src/main/java/frc/robot/poseestimator/PoseEstimatorMath.java

```diff
@@ -0,0 +1,94 @@
+package frc.robot.poseestimator;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Transform2d;
+import edu.wpi.first.math.geometry.Twist2d;
+import frc.robot.poseestimator.observations.VisionObservation;
+
+public class PoseEstimatorMath {
+
+	public static Twist2d addGyroToTwist(Twist2d twist, Rotation2d currentGyroAngle, Rotation2d lastGyroAngle) {
+		boolean hasGyroUpdated = currentGyroAngle != null;
+		if (hasGyroUpdated) {
+			return updateChangeInAngle(twist, currentGyroAngle, lastGyroAngle);
+		}
+		return twist;
+	}
+
+	public static Twist2d updateChangeInAngle(Twist2d twist, Rotation2d currentGyroAngle, Rotation2d lastGyroAngle) {
```

Comment:
Same here -- update, means changing something

Reply:
no need

Reply:
it is being changed, it returns a twist with the same values as before except the angle, that counts as an update